### ISSUE TITLE
How to make graph in Calibration tutorial?

### ISSUE THREAD

**Issue by Patrickyp** [kriswiner/MPU9250#273]
I am looking at this tutorial and I don't know how the dotted graphs were made.  What exactly is the format of the magnetometer to print to serial monitor and get that graphed data?  Any example code?  

tutorial:
https://github.com/kriswiner/MPU6050/wiki/Simple-and-Effective-Magnetometer-Calibration

Thanks.

**Comment by kriswiner** [comment]
Capture mx, my,mz to serial monitor, import into spreadsheet, plot mx vs
my, mx vs. mz, my vs. mz.

**Comment by Patrickyp** [comment]
Thanks for your answer.  So should I be copying your magcalMPU9250() function in to my code?  I am running MPU9250BasicAHRS.ino which is lacking that test but has the function calibrateMPU9250() which I assume alone is insufficient?

**Comment by kriswiner** [comment]
Yes, you wil need some way to calibrate the mags. Why not start with a more
modern sketch altogether like, maybe, this one:

https://github.com/kriswiner/ESP8285/blob/master/MPU9250/MPU9250_MS5637_BasicAHRS2_ESP8266.ino

**Comment by Patrickyp** [comment]
Would MPU9250BasicAHRS.ino work just as well assuming I implement the magcalMPU9250 function correctly?  I am having trouble getting the one you linked to compile as I am not using MS5637 just mpu9250 from sparkfun and a teensy 3.6.

**Comment by kriswiner** [comment]
Maybe, but not the latest...

**Comment by Patrickyp** [comment]
DO you know how often I need to do the figure 8 calibration?  It will be a hassle for the project I'm working on but it seems like I have to do it pretty frequently to get accurate data.

**Comment by kriswiner** [comment]
Just do it once, well, and store the values in the Arduino sketch so you
don;t have to do it again, or only rarely.

**Comment by khoa21214475** [comment]
A stupid question: how big figure 8 is?, i try to get stable pitch roll yaw but i cant...

**Comment by MitalPattani** [comment]
Try using serial plotter instead if just want to check the calibration.

**Comment by khoa21214475** [comment]
I will try, thanks ! 😃

**Comment by zenmanenergy** [comment]
If you get that to work, will you please post a link to the plotter you used and/or a short video of getting it working? I'm struggling with calibration too.

**Comment by kriswiner** [comment]
Stream comma delimited mag data to the serial monitor, grab it and import
into any spreadsheet to plot Mx vs. My, My vs. Mz, Mx vs. Mz, etc.

Not sure why this is so hard...

**Comment by khoa21214475** [comment]
may be i have good chart...
![mag_clib_chart](https://user-images.githubusercontent.com/36263014/43113683-8bd93d00-8f25-11e8-8ab1-1b0b58ed4a2b.png)
Thanks Kriswiner so much !

**Comment by khoa21214475** [comment]
And an additionally question, how can we calibrate accelerate and gyro sensor, put face up on flat and  stand still, right?

**Comment by kriswiner** [comment]
Yes, or with a little more effort like this
<https://github.com/kriswiner/EM7180_SENtral_sensor_hub/wiki/F.--Magnetometer-and-Accelerometer-Calibration>
.

**Comment by khoa21214475** [comment]
I can't buy em7180 + mpu9250 + bmp280 😂.
2 degree with this board, and how about the accuracy without em7180 and bmp280 ?

**Comment by kriswiner** [comment]
" I can't buy em7180 + mpu9250 + bmp280 "  Why not?

Without EM7180 you could get 4 degree heading accuracy when well
calibrated. See this
<https://github.com/kriswiner/EM7180_SENtral_sensor_hub/wiki/K.-Limits-of-Absolute-Heading-Accuracy-Using-Inexpensive-MEMS-Sensors>
.

**Comment by khoa21214475** [comment]
Thanks for quickly replies Kriswiner!
I will try to get one module with ESP32 + GPS Neo 6M + mpu9250 + bmp280 + em7180 , i want to build this module for tracking celestial body.
Im Ok to get Az, Alt from module GPS Neo 6M, and now i will use two information to control step motor by using mpu9250 + bmp280 + em7180.
But its not easy to get minimum accuracy with mpu9250 😅
